### PR TITLE
feat(evm): Raw Ethereum transaction relay support

### PR DESCRIPTION
**Pull request by ilblackdragon**
Adds support for relaying signed transactions from Ethereum.
Can be used by MetaMask configured to the right RPC proxy or NEAR's Web3 provider in `sendRawTransaction` method.

Addresses few collateral issues:
  - converting `chain_id` to u64, as prior it was u128 which didn't match OE EVM interface
  - fixing `view_call` function when it's used for contract deployment (this is used for dry_run)

Test plan
---
Adding unit test for sending signed transactions: transfer, deploy code and function call. Testing invalid signature and chain_id passes inside the transaction.
Adding support for sendRawTransaction in near-web3-provider repo: https://github.com/near/near-web3-provider/pull/70

### CODE REVIEW

File: runtime/near-evm-runner/src/runner.rs
Comment (bowenwang1996) [inline runtime/near-evm-runner/src/runner.rs:307]: what exactly is wrong?

File: runtime/near-evm-runner/src/types.rs
Comment (bowenwang1996) [inline runtime/near-evm-runner/src/types.rs:156]: is it somehow guaranteed that `self.v >= 35`?

**Review by bowenwang1996 — COMMENTED** [review]
what's the reason for changing the type of `chain_id` from `u128` to `u64`?

File: runtime/near-evm-runner/src/runner.rs
Comment (ilblackdragon) [reply runtime/near-evm-runner/src/runner.rs:307]: because the attached deposit coming from the relayer. Instead value should be attached by the meta call itself, charging the user that signed meta tx.
I think we are missing the `value` field in the meta call and should not allow to attach funds by relayers

File: runtime/near-evm-runner/src/types.rs
Comment (ilblackdragon) [reply runtime/near-evm-runner/src/types.rs:156]: the link above defines what `v` can be: 27/28 or 35 + chain_id*2 + {0, 1}

File: runtime/near-evm-runner/src/runner.rs
Comment (ailisp) [reply runtime/near-evm-runner/src/runner.rs:307]: parse_meta_call returns fee value as fee_amount now:
```
    Ok(MetaCallArgs { sender, nonce, fee_amount, fee_address, contract_address, input })

```

**Review by ailisp — APPROVED** [review]

**Comment by ilblackdragon** [comment]
@bowenwang1996 because `chain_id` is u64: https://github.com/near/nearcore/blob/master/runtime/near-evm-runner/src/near_ext.rs#L74 
It was my original mistake to make it u128 everywhere in our code.

File: runtime/near-evm-runner/src/runner.rs
Comment (ilblackdragon) [reply runtime/near-evm-runner/src/runner.rs:318]: That's not the `value`.
`Value` is how much signer attached of base token with the call.
This considers attached deposit coming from relayer, which is incorrect.
We need to add another field to the meta call - `value` (or can call it attached_deposit or whatever) that will actually specify how much to deduct from given user (if they have enough balance).
See the implementation of raw_call for example of such deduction.

Comment (ailisp) [reply runtime/near-evm-runner/src/runner.rs:318]: Got it, some subword of `attached_(base_token)_deposit_value` :D

**Comment by artob** [comment]
To be pedantic, _"Chain ID is a 256-bit value"_ per [EIP-1344](https://eips.ethereum.org/EIPS/eip-1344). However, in terms of OpenEthereum's APIs, they do only take a 64-bit value. And all [registered chain IDs](https://chainid.network) do fit within 64 bits, and certainly our own chain IDs do, so probably no good reason to need it be larger.

**Comment by artob** [comment]
> @bowenwang1996 because `chain_id` is u64: https://github.com/near/nearcore/blob/master/runtime/near-evm-runner/src/near_ext.rs#L74
> It was my original mistake to make it u128 everywhere in our code.

To be pedantic, _"Chain ID is a 256-bit value"_ per [EIP-1344](https://eips.ethereum.org/EIPS/eip-1344). However, in terms of OpenEthereum's APIs, they do only take a 64-bit value. And all [registered chain IDs](https://chainid.network) do fit within 64 bits, and certainly our own chain IDs do, so probably no good reason to need it be larger.

**Review by artob — APPROVED** [review]
Looks really good overall, and significantly increases our out-of-the-box Eth1 compatibility.

My only comments are some nits that can be addressed in a later iteration.

**Comment by artob** [comment]
@ilblackdragon @ailisp FYI, I've rebased this pull request against the current master branch just now.

File: runtime/near-evm-runner/src/types.rs
Comment (evgenykuzyakov) [inline runtime/near-evm-runner/src/types.rs:113]: Can you please add comments to the fields

File: runtime/near-evm-runner/src/types.rs
Comment (evgenykuzyakov) [inline runtime/near-evm-runner/src/types.rs:141]: Add comments here

**Review by evgenykuzyakov — APPROVED** [review]
You need to bump EVM protocol version to hardfork betanet.

File: runtime/near-evm-runner/src/types.rs
Comment (artob) [reply runtime/near-evm-runner/src/types.rs]: Resolved in 311b3c8ce1c18d1662e5549575f2b3ad8657a21a.

File: runtime/near-evm-runner/src/types.rs
Comment (artob) [reply runtime/near-evm-runner/src/types.rs]: Resolved in 311b3c8ce1c18d1662e5549575f2b3ad8657a21a.

**Comment by artob** [comment]
> You need to bump EVM protocol version to hardfork betanet.

I suggest we merge this at the same time as #3797 and thus only need to bump once (in that PR).

**Review by artob — APPROVED** [review]

**Comment by artob** [comment]
@bowenwang1996 Need your review as code owner to be able to merge this.

**Comment by artob** [comment]
> You need to bump EVM protocol version to hardfork betanet.

Done, just pushed now. @bowenwang1996 This is all good to go as far as @ilblackdragon and I are concerned, if you would have a chance to review.

**Review by bowenwang1996 — APPROVED** [review]

**Comment by artob** [comment]
@olonho @willemneal Could we get a review from you guys as well, pretty please, to unblock merging this?

**Review by willemneal — APPROVED** [review]
LGTM just one logic question.

File: runtime/near-evm-runner/src/runner.rs
Comment (ilblackdragon) [reply runtime/near-evm-runner/src/runner.rs]: Good catch. An artifact of restructuring code.

**Pull request merged**
